scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/syntax/MagnolifySyntax.scala

```diff
@@ -0,0 +1,147 @@
+  Query,
+  Sharding,
+  Table,
+  TimePartitioning
+}
+import com.spotify.scio.coders.Coder
+import com.spotify.scio.io.ClosedTap
+import com.spotify.scio.values.SCollection
+import magnolify.bigquery.TableRowType
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO.Write.{
+  CreateDisposition,
+  Method,
+  WriteDisposition
+}
+import org.apache.beam.sdk.io.gcp.bigquery.InsertRetryPolicy
+import org.joda.time.Duration
+
+final class MagnolifyBigQueryScioContextOps(private val self: ScioContext) extends AnyVal {
+
+  /**
+   * Get an SCollection for a BigQuery SELECT query. Both
+   * [[https://cloud.google.com/bigquery/docs/reference/legacy-sql Legacy SQL]] and
+   * [[https://cloud.google.com/bigquery/docs/reference/standard-sql/ Standard SQL]] dialects are
+   * supported. By default the query dialect will be automatically detected. To override this
+   * behavior, start the query string with `#legacysql` or `#standardsql`.
+   */
+  def typedBigQuerySelect[T: TableRowType: Coder](
+    sqlQuery: Query,
+    flattenResults: Boolean = MagnolifyBigQuerySelect.ReadParam.DefaultFlattenResults
+  ): SCollection[T] =
+    self.read(MagnolifyBigQuerySelect(sqlQuery))(MagnolifyBigQuerySelect.ReadParam(flattenResults))
+
+  /** Get an SCollection for a BigQuery table. */
+  def typedBigQueryTable[T: TableRowType: Coder](table: Table): SCollection[T] =
+    self.read(MagnolifyBigQueryTable(table))
+
+  /**
+   * Get an SCollection for a BigQuery table using the storage API.
+   *
+   * @param selectedFields
+   *   names of the fields in the table that should be read. If empty, all fields will be read. If
+   *   the specified field is a nested field, all the sub-fields in the field will be selected.
+   *   Fields will always appear in the generated class in the same order as they appear in the
+   *   table, regardless of the order specified in selectedFields.
+   * @param rowRestriction
+   *   SQL text filtering statement, similar ti a WHERE clause in a query. Currently, we support
+   *   combinations of predicates that are a comparison between a column and a constant value in SQL
+   *   statement. Aggregates are not supported. For example:
+   *
+   * {{{
+   * "a > DATE '2014-09-27' AND (b > 5 AND c LIKE 'date')"
+   * }}}
+   */
+  def typedBigQueryStorageMagnolify[T: TableRowType: Coder](
+    table: Table,
+    selectedFields: List[String] = MagnolifyBigQueryStorage.ReadParam.DefaultSelectFields,
+    rowRestriction: String = null
+  ): SCollection[T] =
+    self.read(MagnolifyBigQueryStorage(table, selectedFields, Option(rowRestriction)))
+
+  /**
+   * Get an SCollection for a BigQuery SELECT query using the storage API.
+   *
+   * @param query
+   *   SQL query
+   */
+  def typedBigQueryStorageMagnolify[T: TableRowType: Coder](query: Query): SCollection[T] =
+    self.read(MagnolifyBigQueryStorageSelect(query))
+
+}
+
+final class MagnolifyBigQuerySCollectionOps[T: TableRowType](private val self: SCollection[T]) {
```

Comment:
Prefer requiring the implicit on the `saveAsBigQueryTable` method instead of here